src/types/redis_sb_chain.cc

```diff
@@ -0,0 +1,240 @@
+  LockGuard guard(storage_->GetLockManager(), ns_key);
+  SBChainMetadata sb_chain_metadata;
+  rocksdb::Status s = getSBChainMetadata(ns_key, &sb_chain_metadata);
+
+  if (s.IsNotFound()) {
+    rocksdb::Status create_s =
+        createSBChain(ns_key, kBFDefaultErrorRate, kBFDefaultInitCapacity, kBFDefaultExpansion, 0, sb_chain_metadata);
+    if (!create_s.ok()) return create_s;
+  } else if (!s.ok()) {
+    return s;
+  }
+
+  std::vector<std::string> bf_key_list;
+  std::vector<BFMetadata> bf_metadata_list;
+  for (u_int16_t i = 0; i < sb_chain_metadata.n_filters; ++i) {
+    std::string bf_key, bf_meta_key;
+    appendBFSuffix(ns_key, i, &bf_key);
+    appendBFMetaSuffix(bf_key, &bf_meta_key);
+    BFMetadata bf_metadata;
+    s = getBFMetadata(bf_meta_key, &bf_metadata);
+    if (!s.ok()) return s;
+    bf_key_list.push_back(bf_key);
+    bf_metadata_list.push_back(bf_metadata);
+  }
+
+  BFMetadata cur_bf_metadata = bf_metadata_list.back();
+  std::string cur_bf_key = bf_key_list.back();
+  if (cur_bf_metadata.size + 1 > cur_bf_metadata.entries) {
+    return rocksdb::Status::Aborted("filter is full");
+  }
+
+  auto batch = storage_->GetWriteBatchBase();
+  WriteBatchLogData log_data(kRedisBloomFilter, {"insert"});
+  batch->PutLogData(log_data.Encode());
+
+  bool item_exist = false;
+  const auto &item_string = item.ToString();
+
+  // check
+  for (int i = sb_chain_metadata.n_filters - 1; i >= 0; --i) {
+    s = bloomCheckAdd(bf_key_list[i], item_string, ReadWriteMode::MODE_READ, ret);
+    if (ret == 1) {
+      item_exist = true;
+      break;
+    }
+  }
+
+  // insert
+  if (!item_exist) {
+    s = bloomCheckAdd(cur_bf_key, item_string, ReadWriteMode::MODE_WRITE, ret);
+    cur_bf_metadata.size += ret;
+    sb_chain_metadata.size += ret;
+  }
+
+  std::string cur_bf_metadata_bytes, cur_bf_meta_key, sb_chain_metadata_bytes;
+  cur_bf_metadata.Encode(&cur_bf_metadata_bytes);
+  sb_chain_metadata.Encode(&sb_chain_metadata_bytes);
+  appendBFMetaSuffix(cur_bf_key, &cur_bf_meta_key);
+  batch->Put(cur_bf_meta_key, cur_bf_metadata_bytes);
+  batch->Put(metadata_cf_handle_, ns_key, sb_chain_metadata_bytes);
+
+  return storage_->Write(storage_->DefaultWriteOptions(), batch->GetWriteBatch());
+}
+
+rocksdb::Status SBChain::Exist(const Slice &user_key, const Slice &item, int &ret) {
+  std::string ns_key = AppendNamespacePrefix(user_key);
+  LockGuard guard(storage_->GetLockManager(), ns_key);
+  SBChainMetadata sb_chain_metadata;
+  rocksdb::Status s = getSBChainMetadata(ns_key, &sb_chain_metadata);
+  if (s.IsNotFound()) return rocksdb::Status::NotFound("key is not found");
+  if (!s.ok()) return s;
+
+  std::vector<std::string> bf_key_list;
+  for (uint16_t i = 0; i < sb_chain_metadata.n_filters; ++i) {
+    std::string bf_key;
+    appendBFSuffix(ns_key, i, &bf_key);
+    bf_key_list.push_back(bf_key);
+  }
+
+  const auto &item_string = item.ToString();
+  // check
+  for (int i = sb_chain_metadata.n_filters - 1; i >= 0; --i) {
```

Comment:
(As for direction, now we and RedisBloom are searching from largest BloomFilter to smaller BloomFilter, but I wonder  if we can searching it from smallest to largest? We can leave a todo or note here.)